modules/assim.sequential/R/load_data_paleon_sda.R

```diff
@@ -0,0 +1,148 @@
+##' @title load_data_paleon_sda
+##' @name  load_data_paleon_sda
+##' @author Ann Raiho \email{araiho@nd.edu}
+##' 
+##' @param settings    PEcAn SDA settings object
+##' 
+##' @description Load data function for paleon SDA data products
+##' 
+##' @return obs.mean and obs.cov for sda.enkf
+##' @export
+##' 
+load_data_paleon_sda <- function(settings){
+  
+  #### Types of Data needed
+  ## STEPPS Fcomp
+  ## ReFAB biomass
+  ## HF Tree Rings (CHECK)
+  ## Tree Rings from tree ring module
+  ## PLS and FIA Biomass Snapshot
+  ## Wish list: eddy covariance
+  
+  library(plyr) #need to load to use .fnc below
+  
+  d <- settings$database$bety[c("dbname", "password", "host", "user")]
+  bety <- src_postgres(host = d$host, user = d$user, password = d$password, dbname = d$dbname)
+  settings$host$name <- "localhost"
+  
+  site <- PEcAn.DB::query.site(settings$run$site$id, bety$con)
+  format_id <- settings$state.data.assimilation$data$format_id
+  
+  input.list <- list()
+  input.id <- list()
+  obvs <- list()
+  
+  var.names <- unlist(sapply(settings$state.data.assimilation$state.variable, 
+                             function(x) {
+                               x$variable.name
+                             }, 
+                             USE.NAMES = FALSE), 
+                      use.names = FALSE)
+  
+  start_date <- settings$state.data.assimilation$start.date
+  end_date   <- settings$state.data.assimilation$end.date
+  
+  obs.mean <- list()
+  obs.cov <- list()
+  
+  start.time <- format(lubridate::ymd(start_date),settings$state.data.assimilation$forecast.time.step)
+  end.time <- format(lubridate::ymd(end_date),settings$state.data.assimilation$forecast.time.step)
+  obs.times <- start.time:end.time
+  
+  
+  for(i in seq_along(format_id)){
+    input.list[[i]] <- db.query(paste("SELECT * FROM inputs WHERE site_id =",site$id ,"  AND format_id = ",format_id[[i]]), bety$con)
+    input.id[[i]] <- input.list[[i]]$id
+    
+    data.path <- PEcAn.DB::query.file.path(input.id[[i]], settings$host$name, bety$con)
+    format_full <- format <- PEcAn.DB::query.format.vars(input.id = input.id[[i]], bety, format.id = NA, var.ids=NA)
+    
+    format$na.strings <- 'NA'
+    
+    # ---- LOAD INPUT DATA ---- #
+    time.row <- format$time.row
+    vars.used.index <- setdiff(seq_along(format$vars$variable_id), format$time.row)
+    
+    print(paste('Using PEcAn.benchmark::load_data.R on format_id',format_id[[i]],'-- may take a few minutes'))
+    obvs[[i]] <- PEcAn.benchmark::load_data(data.path, format, start_year = lubridate::year(start_date), end_year = lubridate::year(end_date), site, vars.used.index, time.row)
```

Comment:
1) Yes, the dropping of time.row needs to be done in the function itself (and then adding the pulling of time.row from the format internally in the function) and then this needs to be change everywhere load_data is called. I've asked others to do this in the past, but it seems to have not happened.
3) I suspect the benchmark workflow is like that because you're often NOT grabbing all of the variables in a file